### PR TITLE
[CAS 364] channel list item features

### PR DESCRIPTION
**Pull request by carterhudson**
https://stream-io.atlassian.net/browse/CAS-364

### Description

Adds features for channel list view items:
- Swipe for options menu
- Unread badge count
- User's last message read status
- Selective bolding in last message preview text for mentioned users and message sender

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelUnreadCountDomainTest.kt [hunk 1]
```diff
@@ -18,20 +16,23 @@ internal class ChannelUnreadCountDomainTest : BaseDisconnectedIntegrationTest()
     fun testUnreadCount() {
         val messages = listOf(data.message1, data.message2Older)
```
Comment (carterhudson) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelUnreadCountDomainTest.kt:17]: Should I delete this file?

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt [hunk 2]
```diff
@@ -134,6 +136,14 @@ internal class ChannelControllerImpl(
         .map { it.values.sortedBy { user -> user.createdAt } }
         .asLiveData()
 
+    val _typingUsers: StateFlow<List<User>> = _typing.map {
+        it.values
+            .sortedBy(ChatEvent::createdAt)
+            .mapNotNull { event ->
+                (event as? TypingStartEvent)?.user ?: (event as? TypingStopEvent)?.user
```
Comment (JcMinarro) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:143]: If the event is `TypingStopEvent` it indicate that the user has stopped to write, Shouldn't it be removed?

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelUnreadCountDomainTest.kt (hunk 1 above)
Comment (JcMinarro) [reply stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelUnreadCountDomainTest.kt:17]: Isn't those tests useful?

Comment (carterhudson) [reply stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelUnreadCountDomainTest.kt:17]: I reworked `computeUnread` and wrote tests for it in `ComputeUnreadCountTest`

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt (hunk 2 above)
Comment (carterhudson) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:143]: Thierry wrote this and I forgot about it. I'll check this out.

Comment (carterhudson) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt:143]: I think Thierry changed this. I'll check it out.

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**